tests/openvino/test_quantization.py

```diff
@@ -589,3 +594,38 @@ def compute_metrics(p):
             tokens = tokenizer("This is a sample input", return_tensors="pt")
             outputs = model(**tokens)
             self.assertTrue("logits" in outputs)
+
+
+class InferRequestWrapperTest(unittest.TestCase):
+    MODEL_ID = ("openai/whisper-tiny.en",)
+
+    @staticmethod
+    def _generate_random_audio_data(processor):
+        t = np.linspace(0, 1.0, int(1000), endpoint=False)
+        audio_data = 0.5 * np.sin((2 + np.random.random()) * np.pi * t)
+        input_features = processor(
+            audio_data,
+            sampling_rate=16000,
+            return_tensors="pt",
+        ).input_features
+        return input_features
+
+    @parameterized.expand(MODEL_ID)
+    def test_calibration_data_uniqueness(self, model_id):
+        ov_model = OVModelForSpeechSeq2Seq.from_pretrained(model_id, export=True, compile=True)
+        processor = AutoProcessor.from_pretrained(model_id)
+
+        calibration_data = []
+        ov_model.decoder_with_past.request = InferRequestWrapper(ov_model.decoder_with_past.request, calibration_data)
+        for _ in range(2):
+            input_features = self._generate_random_audio_data(processor)
+            ov_model.generate(input_features)
+
+        data_hashes_per_key = defaultdict(list)
+        for inputs_dict in calibration_data:
+            for k, v in inputs_dict.items():
+                x = (v.numpy() if isinstance(v, torch.Tensor) else v).copy()
+                data_hashes_per_key[k].append(hash(x.tobytes()))
+        for k, data_hashes in data_hashes_per_key.items():
+            # All hashes can not be equal because calibration dataset contains at least 2 different samples
+            assert any(data_hashes[0] != it for it in data_hashes), f"Collected tensors are all equal for input {k}"
```

Comment:
can you please replace it with 
```suggestion
            self.assertTrue(any(data_hashes[0] != it for it in data_hashes))
```
also shouldn't it be :
```suggestion
            self.assertTrue(len(set(data_hashes)) == num_samples)
```
with here num_samples = 2 ?

Reply:
> can you please replace it with

Replaced

> also shouldn't it be :

This condition wouldn't hold because during a single whisper model inference, the decoder model is inferenced multiple times autoregressively. So in total the length of `calibration_data` list is larger than the number of used audio samples, in this case 2. Moreover, for some input dict keys there are more unique tensors collected than for others so there's no single `num_samples` value for all of them. That's why I just check that they are not all equal for some key.